### DISCUSSION TITLE
Understanding PDM init Python interpreter selection

### DISCUSSION THREAD

**Discussion by robml** [pdm-project/pdm#971] [General]
So I love PDM, and want to slowly understand a bit more behind the scenes. Every time I initiate a project through `pdm init` I get the following options (note I only have Python3.8 on my system after removing the others):

```
Creating a pyproject.toml for PDM...
Please enter the Python interpreter to use
0. /usr/bin/python3.8 (3.8)
1. /usr/bin/python2.7 (2.7)
2. /home/robmel/.local/share/pdm/venv/bin/python (3.8)
Please select: [0]:
```
Now usually I would select `0` and call it a day, but I am curious what that last option: `/home/robmel/.local/share/pdm/venv/bin/python (3.8)` is, and why it's created? Also does it just reference the system python interpreter at the time of initiation or during all consequent visits as well? So far I have been using `/usr/bin/python3.8` because the explicit declaration of `python3.8` makes me more comfortable but curious what the other option is.

**Comment by frostming** [comment]
Because you have installed PDM via the install-pdm.py script, which creates a venv at the particular location of 2. for you. Then, it gets picked up as `sys.executable` inside the PDM process.
